rust/extensions.bzl

```diff
@@ -40,18 +51,25 @@ def _rust_impl(module_ctx):
     toolchains = root.tags.toolchain or rules_rust.tags.toolchain
 
     for toolchain in toolchains:
-        rust_register_toolchains(
-            dev_components = toolchain.dev_components,
-            edition = toolchain.edition,
-            allocator_library = toolchain.allocator_library,
-            rustfmt_version = toolchain.rustfmt_version,
-            rust_analyzer_version = toolchain.rust_analyzer_version,
-            sha256s = toolchain.sha256s,
-            extra_target_triples = toolchain.extra_target_triples,
-            urls = toolchain.urls,
-            versions = toolchain.versions,
```

Comment:
Why not modify the implementation of `rust_register_toolchains` to not register any toolchains when passed `versions = []`? This way we avoid the extra check and _empty_repository setup.

Reply:
There did not seem to be a clean way to get an empty list passed as `toolchain_names` to `toolchain_repository_hub` here (https://github.com/bazelbuild/rules_rust/blob/main/rust/repositories.bzl#L252) without nesting most of that function under an `if len(versions) > 0:`.   `versions` is not considered until inside calls to `rust_repository_set` and `_get_toolchain_repositories`.   I could add the following at the beginning of `rust_register_toolchains` to avoid having to declare `_empty_repository`: 

```
if len(versions) == 0:
    toolchain_repository_hub(
        name = "rust_toolchains",
        toolchain_names = [],
        toolchain_labels = {},
        toolchain_types = {},
        exec_compatible_with = {},
        target_compatible_with = {},
    )
```

Other refactor suggestions are welcome.

Reply:
Thanks! I think what you have now it cleaner than the `toolchain_repository_hub` approach.